[FEAT] Unify method for generating filepath to check map existance and print out the filepath

hdevine825 opened the pull request:
## This PR…

- Resolves an issue where a recent map data file would be downloaded again if the country name includes a slash.

## Considerations and implementations

Used the existing function `build_osm_pbf_filepath` to create the file path to check against instead of the existing method that resulted in checking in a subfolder when a slash was included.

## How to test

1. Run with a -co value that includes a slash like us/delaware
2. After download rerun with the same value and the file won't be downloaded again

## Pull Request Checklist
- [x] Reviewed the [Contributing Guidelines](https://github.com/treee111/wahooMapsCreator/blob/develop/docs/CONTRIBUTING.md)
    + Especially the [Pull-Requests](https://github.com/treee111/wahooMapsCreator/blob/develop/docs/CONTRIBUTING.md#Pull-Requests) section
- [x] Commits (and commit-messages) are understandable
- [x] Tested with macOS / Linux
- [ ] Tested with Windows

hdevine825 committed:
Change method for creating filepath to check map data age

treee111 commented:
Hi @hdevine825, thanks for your PR 👍 

I wonder how you even came this far :D

If I run `python -m wahoomc cli -co delaware`

I get the printed error due to not resolving us/delaware and us_delaware accordingly.
Don't you have this error? How did you call wahooMapsCreator for your generations or us/delaware?

```
(gdal-dev)  wahooMapsCreator % python -m wahoomc cli -co delaware
INFO:--------------------------------------------------------------------------------
INFO:--------------------------------------------------------------------------------
INFO:# check geofabrik.json file
INFO:# Input country: delaware.
INFO:--------------------------------------------------------------------------------
INFO:# Determine involved/border countries
INFO:+ Involved country: us_delaware
INFO:+ Involved country: us_maryland
INFO:+ Involved country: us_new-jersey
INFO:+ Involved country: us_pennsylvania
INFO:+ Involved country: us_virginia
INFO:+ Border countries will be processed
INFO:--------------------------------------------------------------------------------
INFO:# check land_polygons.shp file
INFO:--------------------------------------------------------------------------------
INFO:# check countries .osm.pbf files
INFO:+ Checking for old maps and remove them
INFO:--------------------------------------------------------------------------------
INFO:# Downloading us_delaware-latest.osm.pbf file
Traceback (most recent call last):
  File "/opt/anaconda3/envs/gdal-dev/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/anaconda3/envs/gdal-dev/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/wahooMapsCreator/wahoomc/__main__.py", line 7, in <module>
    main.run("run")
  File "/Users/wahooMapsCreator/wahoomc/main.py", line 73, in run
    o_downloader.download_files_if_needed()
  File "/Users/wahooMapsCreator/wahoomc/downloader.py", line 289, in download_files_if_needed
    self.download_osm_pbf_file()
  File "/Users/wahooMapsCreator/wahoomc/downloader.py", line 370, in download_osm_pbf_file
    download_file(map_file_path, url)
  File "/Users/wahooMapsCreator/wahoomc/downloader.py", line 50, in download_file
    if url.split('.')[-1] == 'zip':
AttributeError: 'NoneType' object has no attribute 'split'
```

hdevine825 commented:
I will have to go back and re-understand what I did. Probably some time next month... Feel free to close the PR for now if you want to tidy up.

treee111 committed:
restructure to not have a IndexError

- IndexError occurred when map file was not downloaded already and hence the log.debug printed a not-existing part of the array/dict/list

treee111 commented:
no worries!
I made this PR #276 and was able to run wahooMapsCreator with `delaware` and `us/delaware`.

On both processings I was not able to see the behaviour that countries are downloaded if they already exist. The slash should not play a role here, since internally we work without slash since very early in the processing of wahooMapsCreator. See the logging. With the mentioned PR I used the slash version of the country in one location to get the download URL.

However to use existing logic for the same reason (`build_osm_pbf_filepath`) makes very much sense and also outputting the location where existing maps are searched for as well!

I make a change to work also when the file does not exist locally and will merge this accordingly.

Thanks for this! 👍
